gossip/src/crds_value.rs

```diff
@@ -163,6 +170,110 @@ impl CrdsValue {
         self.data.pubkey()
     }
 
+    // Implements bincode::serialize_into for CrdsValue.
+    pub(crate) fn bincode_serialize<W: Write>(&self, writer: &mut W) -> Result<(), IoError> {
+        writer.write_all(self.signature.as_ref())?;
+        writer.write_all(&self.bincode_serialized_data)
+    }
+
+    // Implements bincode::serialize_into for Vec<CrdsValue>.
+    pub(crate) fn bincode_serialize_many<I, T: Borrow<Self>, W: Write>(
+        values: I,
+        writer: &mut W,
+    ) -> Result<(), IoError>
+    where
+        I: IntoIterator<Item = T>,
+        <I as IntoIterator>::IntoIter: ExactSizeIterator,
+    {
+        let mut values = values.into_iter();
+        let size = u64::try_from(values.len()).unwrap();
+        writer.write_all(&size.to_le_bytes())?;
+        values.try_for_each(|value| value.borrow().bincode_serialize(writer))
+    }
+
+    // Implements bincode::deserialize_from for CrdsValue.
+    pub(crate) fn bincode_deserialize(
+        bytes: &[u8],
+        allow_trailing_bytes: bool,
```

Comment:
can you explain trailing bytes? Not sure I understand why we would allow these?

Reply:
It is the same thing as in bincode:
https://docs.rs/bincode/1.3.3/bincode/config/trait.Options.html#method.allow_trailing_bytes

If you are only deserializing a single CrdsValue then generally you do not want to allow trailing bytes.
But you may be deserializing a `struct Foo(CrdsValue, Bar)` or a `Vec<CrdsValue>` in which case the trailing bytes are the next value to be deserialized, so you would want to allow trailing bytes.